pkg/client/openstack/kluster/client.go

```diff
@@ -137,31 +137,26 @@ func (c *klusterClient) RebootNode(id string) error {
 }
 
 func (c *klusterClient) ListNodes(k *v1.Kluster, pool *models.NodePool) ([]Node, error) {
-	var unfilteredNodes []Node
 	var filteredNodes []Node
 
-	err := servers.List(c.ComputeClient, servers.ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+	err := servers.List(c.ComputeClient, servers.ListOpts{Name: "^(kks-)?" + k.Spec.Name + "-" + pool.Name + "-"}).EachPage(func(page pagination.Page) (bool, error) {
```

Comment:
@joker-at-work Is using a regex for listing servers by name safe and won't break in the future in your opionen? Seems like its depended on the database used for nova which unsettles me a bit.

Reply:
All I can say is that we don't plan to change the DB under Nova once again. 

FYI this is the function translating the string into the query: https://github.com/openstack/nova/blob/master/nova/db/sqlalchemy/api.py#L1943-L1973 Looks pretty DB-dependent. It basically just uses a sanitized version of whatever you provide.

Reply:
Ok thanks.